phonebook-api/src/test/java/com/telran/phonebookapi/repository/ConfirmationTokenRepositoryTest.java

```diff
@@ -0,0 +1,53 @@
+package com.telran.phonebookapi.repository;
+
+import com.telran.phonebookapi.entity.ConfirmationToken;
+import com.telran.phonebookapi.entity.User;
+import org.junit.jupiter.api.Test;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.autoconfigure.orm.jpa.DataJpaTest;
+import org.springframework.boot.test.autoconfigure.orm.jpa.TestEntityManager;
+import org.springframework.transaction.annotation.Transactional;
+
+import java.util.Optional;
+
+import static org.junit.jupiter.api.Assertions.*;
+
+@DataJpaTest
+class ConfirmationTokenRepositoryTest {
+
+    @Autowired
+    ConfirmationTokenRepository tokenRepository;
+
+    @Autowired
+    TestEntityManager entityManager;
+
+    @Test
+    @Transactional
+    public void testFindByToken_oneTokenInDb_oneFound() {
+        User user = new User("anna@gmail.com", "sdfafdfdfrdf");
+        ConfirmationToken token = new ConfirmationToken(user);
+
+        entityManager.persist(user);
+        entityManager.persist(token);
+        entityManager.flush();
```

Comment:
add "em.clear()" after flush. It removes persistent context and forces ORM (Hibernate) to appeal to the DB